bevy_matchbox/src/signaling.rs

```diff
@@ -0,0 +1,161 @@
+use std::net::SocketAddr;
+
+use bevy::{
+    ecs::system::Command,
+    prelude::{Commands, Resource},
+    tasks::IoTaskPool,
+};
+pub use matchbox_signaling;
+use matchbox_signaling::{
+    topologies::{
+        client_server::{ClientServer, ClientServerCallbacks, ClientServerState},
+        full_mesh::{FullMesh, FullMeshCallbacks, FullMeshState},
+        SignalingTopology,
+    },
+    SignalingCallbacks, SignalingServer, SignalingServerBuilder, SignalingState,
+};
+
+/// A [`SignalingServer`] as a [`Resource`].
+///
+/// As a [`Resource`], with [`Commands`]
+/// ```
+/// use std::net::Ipv4Addr;
+/// use bevy_matchbox::{
+///     prelude::*,
+///     matchbox_signaling::topologies::full_mesh::{FullMesh, FullMeshState}
```

Comment:
Not 100% sure I like this import path, should we re-export `typologies`?

Reply:
I guess we could at least re-export with the matchbox prefixes removed, so

bevy_matchbox::signaling::topologies:full_mesh::FullMesh

But maybe it would make sense to export some of those higher up within matchbox_signaling... What do you think @simbleau ?

Reply:
I'm good with anything, currently operating my own bevy matchbox-like solution which is too hairy and custom at this point to roll into a PR.

That being said, if bevy_matchbox becomes more useful in the future I want to switch. So I support the dev here. :)

Reply:
Yeah, it would be nice if we could share our lessons learned.

I did my best to summarize my implementation here: https://github.com/johanhelsing/matchbox/issues/216

I'd be really impressed if someone made a PR out of it, but hopefully it conveys some better architectural suggestions or returns me some net-positive feedback.